guides/hosting/github.md

```diff
@@ -0,0 +1,124 @@
+# Hosting on GitHub
+
+- Create a project with the same `name` and `description` as specified in your `shard.yml`.
+
+- Add and commit everything:
+```bash
+$ git add -A && git commit -am "shard complete"
+```
+- Add the remote: (Be sure to replace `<YOUR-GITHUB-USERNAME>` and `<YOUR-REPOSITORY-NAME>` accordingly)
+
+NOTE: If you like, feel free to replace `public` with `origin`, or a remote name of your choosing.
+```bash
+$ git remote add public https://github.com/<YOUR-GITHUB-NAME>/<YOUR-REPOSITORY-NAME>.git
+```
+- Push it:
+```bash
+$ git push public master
+```
+
+#### GitHub Releases
+It's good practice to do GitHub Releases.
+
+Add the following markdown build badge below the description in your README to inform users what the most current release is:
+(Be sure to replace `<YOUR-GITHUB-USERNAME>` and `<YOUR-REPOSITORY-NAME>` accordingly)
+
+```Markdown
+[![GitHub release](https://img.shields.io/github/release/<YOUR-GITHUB-USERNAME>/<YOUR-REPOSITORY-NAME>.svg)](https://github.com/<YOUR-GITHUB-USERNAME>/<YOUR-REPOSITORY-NAME>/releases)
+```
+
+Start by navigating to your repository's _releases_ page.
+  - This can be found at `https://github.com/<YOUR-GITHUB-NAME>/<YOUR-REPOSITORY-NAME>/releases`
+
+Click "Create a new release".
+
+According to [the Crystal Shards README](https://github.com/crystal-lang/shards/blob/master/README.md),
+> When libraries are installed from Git repositories, the repository is expected to have version tags following a semver-like format, prefixed with a `v`. Examples: v1.2.3, v2.0.0-rc1 or v2017.04.1
+
+Accordingly, in the input that says `tag version`, type `v0.1.0`. Make sure this matches the `version` in `shard.yml`. Title it `v0.1.0` and write a short description for the release.
+
+Click "Publish release" and you're done!
+
+You'll now notice that the GitHub Release badge has updated in your README.
+
+Follow [Semantic Versioning](http://semver.org/) and create a new release every time your push new code to `master`.
+
+### Travis CI and `.travis.yml`
+If you haven't already, [sign up for Travis CI](https://travis-ci.org/).
+
+Insert the following markdown build badge below the description in your README.md:
+(be sure to replace `<YOUR-GITHUB-USERNAME>` and `<YOUR-REPOSITORY-NAME>` accordingly)
+```Markdown
+[![Build Status](https://travis-ci.org/<YOUR-GITHUB-USERNAME>/<YOUR-REPOSITORY-NAME>.svg?branch=master)](https://travis-ci.org/<YOUR-GITHUB-USERNAME>/<YOUR-REPOSITORY-NAME>)
+```
+Build badges are a simple way to tell people whether your Travis CI build passes.
+
+Add the following lines to your `.travis.yml`:
```

Comment:
Shouldn't we add these lines to the default generated `.travis.yml`?

Reply:
I'm not sure what to do with this comment. 😕

Reply:
Add the lines to https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/tools/init/template/travis.yml.ecr which is a template auto-generated when you create a new crystal project using `crystal init ...`

Reply:
Fair enough, I guess I meant I don't know what to do in the scope of this pull request.